lib/dal/src/func/authoring.rs

```diff
@@ -175,7 +231,30 @@ impl FuncAuthoringClient {
         code: Option<String>,
         associations: Option<FuncAssociations>,
     ) -> FuncAuthoringResult<()> {
-        save::save_func(ctx, id, display_name, name, description, code, associations).await
+        let func = Func::get_by_id_or_error(ctx, id).await?;
+
+        Func::modify_by_id(ctx, func.id, |func| {
+            func.display_name = display_name.to_owned();
+            func.name = name.to_owned();
+            func.description = description.to_owned();
+            func.code_base64 = code
+                .as_ref()
+                .map(|code| general_purpose::STANDARD_NO_PAD.encode(code));
+
+            Ok(())
+        })
+        .await?;
+
+        if let Some(associations) = associations {
+            let update_associations_start = tokio::time::Instant::now();
+            save::update_associations(ctx, &func, associations).await?;
+            debug!(%func.id, %func.kind,
+                "updating associations took {:?}",
+                update_associations_start.elapsed()
+            );
```

Comment:
I have a suspicion this is taking too long. I see long time loads when attaching some funcs. Going to track this down in another PR.

cc @stack72 since you might be interested